src/averageAges.js

```diff
@@ -14,8 +14,26 @@
  *
  * @return {number}
  */
+
+const isMale = (person) => person.sex === 'm';
+const isFemale = (person) => person.sex === 'f';
+const calculateAge = (person) => person.died - person.born;
+const isMother = (mother, child) => mother.name === child.mother;
+const averageAgeCalculate = (ages) =>
+  ages.reduce((sum, age) => sum + age, 0) / ages.length;
+
 function calculateMenAverageAge(people, century) {
-  // write code here
+  const men = people.filter(person => isMale(person));
+
+  const totalManAges = century
+    ? men.filter(man => Math.ceil(man.died / 100) === century)
+      .map(man => calculateAge(man))
+    : men.map(man => calculateAge(man));
+
+  const aveAgesMen = averageAgeCalculate(totalManAges);
+
+  return aveAgesMen;
```

Comment:
you can skip variable declaration when you return it right next line - return right away
```suggestion
  return averageAgeCalculate(totalManAges);
```